File: back-product-mgmt/Infrastructure/Migrations/DatabaseMigrator.cs
using Npgsql;

namespace Infrastructure.Migrations;

public class DatabaseMigrator
{
    private readonly string _connectionString;

    public DatabaseMigrator(string connectionString)
    {
        _connectionString = connectionString;
    }

    public void MigrateUp()
    {
        using var connection = new NpgsqlConnection(_connectionString);
        connection.Open();

        string script = @"CREATE TABLE IF NOT EXISTS Products(Uuid UUID PRIMARY KEY, Data JSONB NOT NULL);";

        using var command = new NpgsqlCommand(script, connection);
        command.ExecuteNonQuery();

        Console.WriteLine("Table product added to database schema successfully.");
    }

    public void MigrateDown()
    {
        using var connection = new NpgsqlConnection(_connectionString);
        connection.Open();

        string script = @"DROP TABLE Products;";

        using var command = new NpgsqlCommand(script, connection);
        command.ExecuteNonQuery();

        Console.WriteLine("Table Products dropped to database schema successfully.");
    }

}

File: back-order-mgmt/src/infrastructure/migrations/DatabaseMigrator.ts
import type { Pool } from "../../../node_modules/@types/pg/index.js";

export class DatabaseMigrator {
    constructor(private readonly pool : Pool) {}

    async migrateUp() : Promise<void>
    {
        const script = `
            CREATE TABLE IF NOT EXISTS Orders (
                Uuid UUID PRIMARY KEY,
                Data JSONB NOT NULL
            );
        `;

        try {
            await this.pool.query(script);
            console.log("Migration: 'Orders' table created successfully.");
        } catch (error) {
            console.error("Error during migration (migrateUp).");
            throw error;
        }
    }

    async migrateDown() : Promise<void>
    {
        const script = `DROP TABLE IF EXISTS Orders;`;
        try {
            await this.pool.query(script);
            console.log("Migration: 'Orders' table dropped successfully.");
        } catch (error) {
            console.error("Error during migration (migrateDown).");
            throw error;
        }
    }
}